### PR TITLE
WT-13049 Forbid reading the metadata using the checkpoint cursor transaction

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/session/session_dhandle.c [hunk 1]
```diff
@@ -554,6 +554,22 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
      */
 
     is_hs = strcmp(uri, WT_HS_URI) == 0;
+
+    /*
+     * We have already pinned the history store checkpoint dhandle when we open the checkpoint
+     * cursor. No need to resolve and open it again. We only need to lock it here. It is also wrong
+     * to read the metadata with the checkpoint transaction.
+     */
+    if (is_hs && session->hs_checkpoint != NULL) {
```
Comment (etienneptl) [inline src/session/session_dhandle.c:563]: Is it the right place for this fast path? The checks right after on the checkpoint name should still be done.

Comment (quchenhao) [reply src/session/session_dhandle.c:563]:
```
if (is_reserved_name && (!is_hs || session->hs_checkpoint == NULL))
```

The following only applies it is not history store.

File: src/session/session_dhandle.c [hunk 2]
```diff
@@ -554,18 +554,33 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
...
+        ret = __wt_session_get_dhandle(session, uri, session->hs_checkpoint, cfg, flags);
```
[the remark this answers is not in the document]
Comment (kommiharibabu) [reply src/session/session_dhandle.c:564]: Thanks, understood.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by etienneptl — APPROVED** [review]
LGTM!

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**